index.js

```diff
@@ -26,6 +26,20 @@ async function fastifyView (fastify, opts) {
   const defaultCtx = opts.defaultContext
   const globalLayoutFileName = opts.layout
 
+  /**
+   * @type {Map<string, Promise>}
+   */
+  const readFileMap = new Map()
+
+  function readFileSemaphore (filePath) {
+    if (readFileMap.has(filePath) === false) {
+      const promise = readFile(filePath, 'utf-8')
+      readFileMap.set(filePath, promise)
+      return promise.finally(() => readFileMap.delete(filePath))
```

Comment:
@Uzlopak also made a small change here by returning the `.finally` to the original caller. promises with hanging `.then`, .`catch`, or `.finally` give me an uneasy feeling.

Reply:
Unfortunately, i have no clue if this returning finally is ok, as i use finally very rarely. What happens if the promise rejects, will it still reject? or will it pass, because finally() is a promise itself (like then) and so we lose the error case?

Reply:
The promise will still reject as expected, the code in the finally block will run in the next tick of the event loop. 

This case is covered by existing tests, ie test-ejs.js:135 tries to read a non-existing file.

Reply:
ok, I trust you on this one.